### PR TITLE
Added better typing of the `I18n.t` function based on the English translation file

### PR DESCRIPTION
**Pull request by crycode-de**
**PR Checklist:**

-   [x] Provide a meaningful description to this PR or mention which issues this fixes.
-   [ ] ~Add tests for your change. This includes negative tests (i.e. inputs that need to fail) as well as baseline tests (i.e. how should the directory structure look like?).~
-   [x] Run the test suite with `npm test`
-   [x] If there are baseline changes, review them and make a separate commit for them with the comment "accept baselines" if they are desired changes
-   [x] Ensure the project builds with `npm run build`
-   [ ] ~If you added a required option, also add it to the template creation (`.github/create_templates.ts`)~
-   [x] Add a detailed migration description to `docs/updates` explaining what the user needs to do when manually updating an existing project
-   [x] Add your changes to `CHANGELOG.md` (referencing the migration description and this PR or the issue you fixed)

**Description:**  
When using TypeScript and React UI this will add better types for the translate function `I18n.t` so that TypeScript will only accept words defined in the English translation file.
Doing so you can't forget to add new translations to at least the English translation file and your IDE will make suggestions for autocomplete.

To achieve this we need a separate `i18n.ts` file which imports `@iobroker/adapter-react/i18n` and `./i18n/en.json`, merges the type definitions and reexports the `I18n` class with the updated type.
Since these are only type related changes, this will add no overhead at runtime.

### CODE REVIEW

**Comment by AlCalzone** [comment]
Good idea! Have you considered making the change in a `.d.ts` file instead? 
This way we should be able to overwrite the type declarations of `@iobroker/adapter-react/i18n` without requiring users to change the import (untested):
```
declare module "@iobroker/adapter-react/i18n" {
  // ... your changed declarations
}
```

**Comment by crycode-de** [comment]
Yeah I tried this, but couldn't get it to work properly.
Either the definitions from the `.d.ts` file were not applied or the compiler complaint about "already declared...".
Maybe there is a way, but I couldn't find it.

**Comment by AlCalzone** [comment]
The whole declaration merging in ambient declarations is weird. I'll attempt again, based on your work.

File: CHANGELOG.md [hunk 1]
```diff
@@ -114,7 +117,7 @@
 * (bluefox & AlCalzone) Add email address to MIT license (#232)
 
 ## v1.15.1 (2019-06-07)
-* (AlCalzone) Properly fix build breaks from #193.  
+* (AlCalzone) Properly fix build breaks from #193.
```
Comment (AlCalzone) [inline CHANGELOG.md]: Not sure how you have your editor set up, but please don't remove trailing whitespaces in Markdown.

**Comment by AlCalzone** [comment]
By the way, the import in the file was the problem. Without it, it works as a .d.ts.

**Comment by AlCalzone** [comment]
@UncleSamSwiss anything to add?

File: CHANGELOG.md (hunk 1 above)
Comment (crycode-de) [reply CHANGELOG.md]: Oh sorry, that was not intended.
I've turned it off now for markdown files.

**Review by UncleSamSwiss — APPROVED** [review]
Very nice idea.
I didn't try it for myself, but it looks good.

**Pull request merged**